pandas/tests/test_groupby.py

```diff
@@ -629,6 +629,19 @@ def f(grp):
         e.name = None
         assert_series_equal(result, e)
 
+    def test_groupby_dropna(self):
+        # GH 3729
+        data = [['a', 'b', 12, 12, 12],
+                ['a', np.nan, 12.3, 233., 12],
+                ['b', 'a', 123.23, 123, 1],
+                ['a', 'b', 1, 1, 1.]]
+        df = DataFrame(data, columns=['a', 'b', 'c', 'd', 'e'])
+        result = df.groupby(['a', 'b'], dropna=False).sum()
+        expected = DataFrame([[12.30, 233, 12],
```

Comment:
this needs a lot more testing. for Series, Panel. and several variations of using dropna with datetimes, with multi-indexes.